src/main/java/io/github/aurora/scheduler/offers/HttpOfferSetImpl.java

```diff
@@ -199,70 +200,67 @@ private ScheduleRequest createRequest(ResourceRequest resourceRequest, long star
+            .setConnectionRequestTimeout(this.timeoutMs)
+            .setConnectTimeout(this.timeoutMs)
+            .setSocketTimeout(this.timeoutMs)
+            .build();
+    request.setConfig(requestConfig);
+    request.addHeader("Content-Type", "application/json; utf-8");
+    request.addHeader("Accept", "application/json");
+    request.setEntity(new StringEntity(jsonMapper.writeValueAsString(scheduleRequest)));
+    CloseableHttpResponse response = httpClient.execute(request);
     try {
-      HttpPost request = new HttpPost(this.endpoint.toString());
-      RequestConfig requestConfig = RequestConfig.custom()
-              .setConnectionRequestTimeout(this.timeoutMs)
-              .setConnectTimeout(this.timeoutMs)
-              .setSocketTimeout(this.timeoutMs)
-              .build();
-      request.setConfig(requestConfig);
-      request.addHeader("Content-Type", "application/json; utf-8");
-      request.addHeader("Accept", "application/json");
-      request.setEntity(new StringEntity(gson.toJson(scheduleRequest)));
-      CloseableHttpResponse response = httpClient.execute(request);
-      try {
-        HttpEntity entity = response.getEntity();
-        if (entity == null) {
-          throw new IOException("Empty response from the external http endpoint.");
-        } else {
-          return EntityUtils.toString(entity);
-        }
-      } finally {
-        response.close();
+      HttpEntity entity = response.getEntity();
+      if (entity == null) {
+        throw new IOException("Empty response from the external http endpoint.");
+      } else {
+        return EntityUtils.toString(entity);
       }
     } finally {
-      httpClient.close();
+      response.close();
     }
   }
 
-  List<HostOffer> processResponse(String responseStr) throws IOException {
+  List<HostOffer> processResponse(String responseStr, List<Long> offerSetDiffList)
+      throws IOException {
     // process the response
-    ScheduleResponse response = gson.fromJson(responseStr, ScheduleResponse.class);
+    ScheduleResponse response = jsonMapper.readValue(responseStr, ScheduleResponse.class);
     if (response.error == null || response.hosts == null) {
       LOG.error("Response: " + responseStr);
       throw new IOException("response is malformed");
     }
+    LOG.info("Received " + response.hosts.size() + " offers");
 
     Map<String, HostOffer> offerMap = offers.stream()
             .collect(Collectors.toMap(offer -> offer.getAttributes().getHost(), offer -> offer));
-    List<HostOffer> orderedOffers = new ArrayList<>();
     if (response.error.trim().isEmpty()) {
-      for (String host : response.hosts) {
-        HostOffer offer = offerMap.get(host);
-        if (offer == null) {
-          LOG.warn("Cannot find host " + host + " in the response");
-        } else {
-          orderedOffers.add(offer);
+      List<HostOffer> orderedOffers = response.hosts.stream().map(host -> offerMap.get(host)).
```

Comment:
```suggestion
      List<HostOffer> orderedOffers = response.hosts.stream()
          .map(host -> offerMap.get(host))
          .filter(offer -> offer != null)
          .collect(Collectors.toList());
```

We should match the style of the rest of the project. Break these lines up per call after the stream() call.

Reply:
good point. Fixed. will push.